### DISCUSSION TITLE
Using dotnet pack with projects that use a source generator

### DISCUSSION THREAD

**Discussion by StijnOostdam** [dotnet/roslyn#58862] [Q&A] [answered]
Hi,

I have a project that I want to release as a Nuget package. This project has a source generator referenced which generates a few extra classes. When I compile this in release and inspect the DLL, I see that the generated classes are part of the DLL.

I use the simple command for `dotnet pack`:

```
dotnet pack <path to .csproj> -c release
```

However, when I run the above command on that `.csproj` and then publish the nuget package, I can inspect the DLL that is in the `.nupkg` which then does not contain the generated classes.

I have tried to force `dotnet pack` to not build, but instead use the DLL from the build:

```
dotnet pack <path to .csproj> -o <output path of .nupkg file> /p:OutputPath=<path to folder that contains built DLL> --no-build
```

But the above command gives me the following error:

```
C:\Program Files\dotnet\sdk\6.0.200-preview.21617.4\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.targets(136,5): error NETSDK1085: The 'NoBuild' property was set to true but the 'Build' target was invoked. [SourceGenerator.csproj]
```

So in short, my question is: how do I pack a NuGet package of a project that uses a source generator with the generated classes included?

**Comment by jasonmalinowski** [comment]
Hmm, there shouldn't be anything special here. Are you able to share the contents of your .csproj (we don't need source files)? Do you have any other customizations going on?

**Reply by StijnOostdam** [reply]
Thanks for the reply, I was about to post the csproj details but we found the error. We did have some extra customizations, see the answer

**Comment by StijnOostdam** [comment, marked as answer]
A colleague did some extra work this morning and by adding detailed verbosity to the build output discovered the error:

CSC : warning CS8785: Generator 'SourceGenerator' failed to generate source. It will not contribute to the output and compilation errors may occur as a result. Exception was of type 'FileNotFoundException' with message 'Could not load file or assembly 'Parser, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null'.

Parser is a project that the source generator depends on, but is also used by other projects, so it cannot be part of the source generator. When searching, we found a question with a solution that is working for us:

https://github.com/dotnet/roslyn/discussions/47517#discussioncomment-64145

In short: also reference the Parser project in the project that uses the source generator
